### DISCUSSION TITLE
[Table] [Pivot Table] Change column names in visualizations

### DISCUSSION THREAD

**Discussion by ValentinC-BR** [apache/superset#18463] [Ideas]
## Context & Current state

The column names (for display purposes only) can be changed at table level.

However, depending on the filters applied to the chart, or even on the audience or user job, the column will not have the same meaning.

Having the same name for every visualization is a bit limiting, and/or can be misleading in a few cases.

## Feature Request

In tables and pivot tables, the column names could be configurable, so that each chart could display custom column names.

## UI proposition

Similar to the metrics, clicking on the columns used in "group by" section could open a little panel, allowing the user to change the name (via a pen logo in the top-right corner for instance).

## Additional notes

This new panel could also be the opportunity to change number formatting at the chart level (see [this issue](https://github.com/apache/superset/issues/13311) or [this issue](https://github.com/apache/superset/issues/6924) as examples of what users need).

**Comment by junlincc** [comment]
"In addition to the color schemes, which seem to be the most persistent customization feature, every chart presenting numbers should at a minimum have the option to customize the number format. " we are working on it ;) 
cc @yongjie

@kgabryje is in-place changing column name on click feasible in react-pivottable?

**Comment by zhaoyongjie** [comment]
Hi @ValentinC-BR, It's a common request in some production-ready BI software. I am working on refactor metric to unlock these actions.
